proposal/package-importer.md

```diff
@@ -0,0 +1,260 @@
+resolve the dependency URL using the standard resolution for that environment.
+Once resolved, this URL will be loaded in the same way as any other `file:` URL.
+
+For example, `@use "pkg:bootstrap";` would resolve to the path of a
+library-defined export within the `bootstrap` dependency. In Node, that would be
+resolved within `node_modules`, using the [Node resolution algorithm]. In Dart,
+that would be resolved within `pub-cache`, using [package-config].
+
+[node resolution algorithm]: https://nodejs.org/api/packages.html
+[package-config]: https://pub.dev/packages/package_config
+
+To better understand and allow for testing against the recommended algorithm, a
+[Sass pkg: test] repository has been made with a rudimentary implementation of
+the algorithm.
+
+[Sass pkg: test]: https://github.com/oddbird/sass-pkg-test
+
+### Design Decisions
+
+We could use the `~` popularized by Webpack's `load-sass` format, but this has
+been deprecated since 2021. In addition, since this creates a URL that is
+syntactically a relative URL, it does not make it clear to the implementation or
+the reader where to find the file.
+
+While the Dart Sass implementation allows for the use of the `package:` url
+scheme, a similar standard doesn't exist in Node. We chose the `pkg:` url scheme
+as it clearly communicates to both the user and compiler, and does not have
+known conflicts in the ecosystem.
+
+The `pkg` scheme will not be supported in the browser version of Dart Sass. To
+support a similar functionality, a user would need to ensure that files are
+served, and the loader would need to fetch the URL. In order to follow the same
+algorithm for [resolving a file: URL], we would need to make many fetches. If we
+instead require the browser version to have a fully resolved URL, we negate many
+of this spec's benefits. Users may write their own custom importers to fit their
+needs.
+
+[resolving a file: URL]: ../spec/modules.md#resolving-a-file-url
+
+The `pkg` import loader will be exposed through an opt-in option as it adds the
+potential for file system interaction to `compileString` and
+`compileStringAsync`.
+
+#### Node Resolution Decisions
+
+The current recommendation for resolving packages in Node is to add
+`node_modules` to the load paths. We could add `node_modules` to the load paths
+by default, but that lacks clarity to the implementation and the reader. In
+addition, a file may have access to multiple `node_modules` directories, and
+different files may have access to different `node_modules` directories in the
+same compilation.
+
+There are a variety of methods currently in use for specifying a location of the
+default Sass export for npm packages. For the most part, packages contain both
+JavaScript and styles, and use the `main` or `module` root keys to define the
+JavaScript entry point. Some packages use the `"sass"` key at the root of their
+`package.json`. Others have adopted [conditional exports], which is supported by
+Vite.
```

Comment:
It's not clear to me why Vite in specific is called out here. What exactly is it that they support? Do other bundlers not support it?

Reply:
I replaced this with "Other packages have adopted [conditional exports], largely driven by Vite, which resolves Sass paths using the `"sass"` and the `"style"` custom conditions."